### PR TITLE
2 feature 2 interactive scheduler

### PR DESCRIPTION
**Pull request by nitvishn**
I've finished up the ScheduleEvent use case to functional completion -- the commit messages provide a good picture. Some refactoring needs to happen to clean up code, this is primarily in the interactor. Simplifications will be made once more User methods are implemented.

### CODE REVIEW

File: gradlew.bat [hunk 1]
```diff
@@ -0,0 +1,89 @@
+@rem
```
Comment (chhavi-shah) [inline gradlew.bat]: add all these gradle files to gitignore then commit again

**Review by chhavi-shah — CHANGES_REQUESTED** [review]

**Review by paridhi26 — CHANGES_REQUESTED** [review]
I think it is better to have setupScheduleEvent method from the Main class in a separate class of it's own. Something like SchedulerMainScreen or SchedulerEventView. It is better to not clutter Main class, because if we all add our display methods (that setup the respective screens for our features) to the Main class, it will be too big. This leads to large class code smell, which should be avoided. Apart from that, everything looks good.

**Review by Dawei-He2002 — DISMISSED** [review]

**Review by 1StevenJiang1 — DISMISSED** [review]

File: src/main/java/screens/CheckboxListCellRenderer.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        setBackground(list.getBackground());
+        setForeground(list.getForeground());
+
+        setSelected(isSelected);
+        setEnabled(list.isEnabled());
+
+        return this;
+    }
+
+}
```
Comment (chhavi-shah) [inline src/main/java/screens/CheckboxListCellRenderer.java:28]: beautiful implementation

**Review by paridhi26 — DISMISSED** [review]

File: src/main/java/screens/ScheduleEvent/ScheduleEventMainFrame.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            viewModel.addTagName(tag.getName());
+        }
+        for(Task task: currentUserService.getCurrentUser().getTasks()){
+            viewModel.addTaskName(task.getName());
+        }
+
+        ScheduleEventScreen scheduleEventScreen = new ScheduleEventScreen(controller, viewModel);
+
+        screens.add(scheduleEventScreen, "welcome");
+        cardLayout.show(screens, "register");
```
Comment (chhavi-shah) [inline src/main/java/screens/ScheduleEvent/ScheduleEventMainFrame.java]: change name

File: src/main/java/useCases/ScheduleEvent/ScheduleEventInteractor.java [hunk 4]
```diff
@@ -1,11 +1,118 @@
...
+        ScheduleEventResponseModel response = new ScheduleEventResponseModel(requestModel.getEventName());
 
-//    public Event scheduleEvent(){
-//    }
+        if(gateway.persistData()){
+            return presenter.prepareSuccessView(response);
+        }else{
+            return presenter.prepareFailView("Unable to save your new event.");
+        }
+    }
```
Comment (chhavi-shah) [inline src/main/java/useCases/ScheduleEvent/ScheduleEventInteractor.java:117]: great use of checking different cases. good implementation of clean architecture

**Review by chhavi-shah — DISMISSED** [review]

**Review by kldtntg — DISMISSED** [review]

File: src/test/java/usecases/ScheduleEventTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+//        ScheduleEventInputBoundary interactor = new ScheduleEventInteractor(currentUserService, presenter);
+//        ScheduleEventController controller = new ScheduleEventController(interactor);
+////        ScheduleEventScreen scheduleEventScreen = new ScheduleEventScreen(controller);
+//
+////        screens.add(scheduleEventScreen, "welcome");
+//        cardLayout.show(screens, "register");
+//        application.pack();
+//        application.setVisible(true);
+    }
+}
```
Comment (chhavi-shah) [inline src/test/java/usecases/ScheduleEventTest.java:32]: thanks so much for fixing our project, Vishnu. you are our 207 god

Comment (chhavi-shah) [reply src/test/java/usecases/ScheduleEventTest.java:32]: now we know not to mess with github

**Review by chhavi-shah — DISMISSED** [review]

File: src/test/java/usecases/ScheduleEventTest.java (hunk 5 above)
Comment (chhavi-shah) [reply src/test/java/usecases/ScheduleEventTest.java:32]: still thank you despite committing when you should've waited

**Review by chhavi-shah — DISMISSED** [review]

**Review by kldtntg — DISMISSED** [review]

**Review by 1StevenJiang1 — DISMISSED** [review]

**Review by 1StevenJiang1 — DISMISSED** [review]

File: src/test/java/usecases/ScheduleEventTest.java (hunk 5 above)
Comment (chhavi-shah) [reply src/test/java/usecases/ScheduleEventTest.java:32]: this is amazing. thanks so much Vishnu

**Review by chhavi-shah — DISMISSED** [review]

**Comment by chazzybearr** [comment]
I fixed the merge conflict in Task and User and that seemed to create more problems (?)

**Review by chazzybearr — APPROVED** [review]

File: src/test/java/usecases/ScheduleEventTest.java (hunk 5 above)
Comment (chhavi-shah) [reply src/test/java/usecases/ScheduleEventTest.java:32]: Leon stop making me review this one pull request 5 times. Thank you <3

**Review by chhavi-shah — APPROVED** [review]

**Comment by chazzybearr** [comment]
I'm so sorry Chhavi

**Comment by chhavi-shah** [comment]
Thanks Leon

**Comment by chhavi-shah** [comment]
You are not forgiven

**Comment by chazzybearr** [comment]
I'm literally crying right now

**Comment by chhavi-shah** [comment]
The worst is not over. Stay strong, Productive Potato Sloth <3

**Review by 1StevenJiang1 — APPROVED** [review]

**Pull request merged**